### ISSUE TITLE
User not showing up in baltop.

### ISSUE THREAD

**Issue by mibby** [EssentialsX/Essentials#63]
EssentialsX dev 156

I have a few users that do not show up in baltop. One of which controls the most money on the server. The last page in baltop shows 10449 players, but the userdata folder in Essentials has 10452 UUID files. Is there any way to fix this?

**Comment by Kakifrucht** [comment]
I can confirm this issue.

**Comment by Agazed** [comment]
I can also confirm this.

**Comment by vemacs** [comment]
I can repro this with Spigot-Essentials as well, and recall seeing it in 1.7.10. Probably something with the way the plugin is fundamentally designed.

**Comment by mibby** [comment]
Any ideas for a possible fix?

**Comment by drtshock** [comment]
No

**Comment by Agazed** [comment]
I've noticed that only the users with a lot of money get affected by this. I have 2/21,000 users that don't show up.. And they happen to both be in the top 10 of baltop.

**Comment by mibby** [comment]
Still no fix for this? :(

**Comment by mibby** [comment]
@drtshock Any update to this issue for a potential fix? Baltop consistency is pretty important on my server.

**Comment by mibby** [comment]
Bump.

**Comment by SupaHam** [comment]
Sorry I do not understand this issue. Could you maybe rephrase it, please?

**Comment by mibby** [comment]
Some users do not show up in the `/baltop` ordering of richest to poorest when they are offline. Might be related to being one of the richest on the server as someone in my top 10 doesn't show up.

**Comment by SupaHam** [comment]
@mibby I do not have the resources to debug this issue. Would it be possible if you sent me all your userdata files so i can try to reproduce?

**Comment by Kakifrucht** [comment]
I'm pretty sure that sending all that data containing the IP addresses would be illegal in many countries, at least in mine it is.

**Comment by SupaHam** [comment]
Then strip ip addresses.

**Comment by Sikatsu** [comment]
Same issue here. I have one active player called ''Fenno95'', he is the richest player, however he only shows up in /baltop when he is online. When offline, he doesn't show up in /baltop at all.

**Comment by XXLPoseidon** [comment]
Problem still exists... I'm not listed in /baltop

What files/informations do you need?
I did "essentials cleanup 0 0 0" from console when noone is online so there are just 2 userfiles for now (mine and my 2nd char).

/baltop shows only my 2nd

The next weird thing is, after doing cleanup i still have my money.
Where is the database file with the users in it that Essentials remember?

I created a plain new server with just Vault, Essentials & LuckPerms on my local machine and only my 2nd shows up in /baltop

Maybe a bug with some UUIDs?

**Server:**
git-Spigot-4d2f30f-f1f3355 (MC 1.14.3) (Implementing API version 1.14.3-SNAPSHOT)
**EssentialsX**
2.17.0.0

My archived old 1.8 Server with Essentials,  PEX and many many other plugins shows me in /baltop

**Comment by darbyjack** [comment]
> Problem still exists... I'm not listed in /baltop
> 
> What files/informations do you need?
> I did "essentials cleanup 0 0 0" from console when noone is online so there are just 2 userfiles for now (mine and my 2nd char).
> 
> /baltop shows only my 2nd
> 
> The next weird thing is, after doing cleanup i still have my money.
> Where is the database file with the users in it that Essentials remember?
> 
> I created a plain new server with just Vault, Essentials & LuckPerms on my local machine and only my 2nd shows up in /baltop
> 
> Maybe a bug with some UUIDs?
> 
> **Server:**
> git-Spigot-4d2f30f-f1f3355 (MC 1.14.3) (Implementing API version 1.14.3-SNAPSHOT)
> **EssentialsX**
> 2.17.0.0
> 
> My archived old 1.8 Server with Essentials, PEX and many many other plugins shows me in /baltop

Your issue is related to #2581

**Comment by pop4959** [comment]
Most likely has to do with the essentials.balancetop.exclude permission as darbyjack mentioned.

Edit: Was notified by triagonal that this issue is older than the permission. My bad. Regardless, there is not enough information in order to investigate this, and the last report was in 2016. If this is still a problem, please create a new issue showing this problem on the latest version of Essentials.

**Comment by ragedriver** [comment]
Its a new permission in EssentialsX , its called essentials.balancetop.exclude
If you OPEED yourself or if you have ,,*,, in the Group that u are using,you cant show up in the balance
remove the essentials.balancetop.exclude from the group or deop yourself and make a group without essentials.balancetop.exclude . Hope i helped u!

**Comment by mibby** [comment]
This issue had nothing to do with ``essentials.balancetop.exclude``. It was normal users who were being excluded from the top calculation when they were offline. 

I'm not sure if it is still relevant in this current year and have no idea if it has been fixed. However I have since wiped all my player essential files and haven't heard complaints of this after doing so.

**Comment by GriffinSego** [comment]
I've just seen an instance of it with players who <i>are</i> online. Any updates?

**Comment by darbyjack** [comment]
Are they opped?

**Comment by GriffinSego** [comment]
Nope. Normal players, no pex besides default.
